fe/fe-core/src/main/java/org/apache/doris/analysis/BinaryPredicate.java

```diff
@@ -294,29 +291,39 @@ private Type getCmpType() throws AnalysisException {
             }
         }
 
-        if (canCompareDate(getChild(0).getType().getPrimitiveType(), getChild(1).getType().getPrimitiveType())) {
+        Type t1 = getChild(0).getType();
+        Type t2 = getChild(1).getType();
+
+        if(t1.getPrimitiveType() == PrimitiveType.DATE && t2.getPrimitiveType() == PrimitiveType.DATE) {
+            return Type.DATE;
```

Comment:
Maybe following is better ?
```
if (canCompareDtat()) {
    return t1.type();
}
```

Reply:
When t1=DATE t2=DATETIME, it should be return DATETIME.